src/mono/mono/metadata/debug-mono-ppdb.c

```diff
@@ -443,6 +438,24 @@ mono_ppdb_lookup_location (MonoDebugMethodInfo *minfo, uint32_t offset)
 	return location;
 }
 
+
+MonoDebugSourceLocation *
+mono_ppdb_lookup_location (MonoDebugMethodInfo *minfo, uint32_t offset)
+{
+	MonoPPDBFile *ppdb = minfo->handle->ppdb;
+	MonoImage *image = ppdb->image;
+	MonoMethod *method = minfo->method;
+	if (!method->token)
+		return NULL;
+	return mono_ppdb_lookup_location_internal (image, mono_metadata_token_index (method->token), offset, ppdb);
+}
+
+MonoDebugSourceLocation *
+mono_ppdb_lookup_location_enc (MonoImage *image, int idx, uint32_t offset)
+{
+	return mono_ppdb_lookup_location_internal (image, idx+1, offset, NULL);
```

Comment:
I think we should change `MonoDebugInformationEnc:idx` to could starting from 1, instead of 0.  So you don't have to do all these `+1` adjustments in all the callers.

Just change https://github.com/dotnet/runtime/blob/eaaf43fba6e10239970807e943a64b9638e02206/src/mono/mono/component/hot_reload.c#L1219-L1222 
to

```
   encDebugInfo->idx = i + 1;
```

And add a comment to 
https://github.com/dotnet/runtime/blob/eaaf43fba6e10239970807e943a64b9638e02206/src/mono/mono/metadata/mono-debug.h#L117